internal/gatewayapi/envoyextensionpolicy.go

```diff
@@ -309,19 +305,31 @@ func (t *Translator) translateEnvoyExtensionPolicyForRoute(policy *egv1a1.EnvoyE
 
 	// Apply IR to all relevant routes
 	prefix := irRoutePrefix(route)
-	for _, x := range xdsIR {
-		for _, http := range x.HTTP {
-			for _, r := range http.Routes {
-				// Apply if there is a match
-				if strings.HasPrefix(r.Name, prefix) {
-					r.ExtProcs = extProcs
-					r.Wasms = wasms
+	parentRefs := GetParentReferences(route)
```

Comment:
can we keep the code as is, and add an extra line that retrieves `envoyProxy := route.listener.gateway.envoyproxy` ?

Reply:
it looks like that we still need the `parentRefs` to retrieve listeners. And `envoyproxies` can vary by gateway(parentRef of route) so we only need to `buildExtProcs` once for each gateway(parentRef) rather than once for each listener.